lib/inspect_annocheck.c

```diff
@@ -6,81 +6,204 @@
 #include <stdio.h>
 #include <string.h>
 #include <errno.h>
+#include <err.h>
 #include <assert.h>
+#include <libannocheck.h>
 
 #include "rpminspect.h"
 
+/* Global variables */
 static bool reported = false;
 
-/* Trim workdir substrings from a generated string. */
-static char *trim_workdir(const rpmfile_entry_t *file, char *s)
+/*
+ * Try to map the product release string to an appropriate
+ * libannocheck profile.  This is going to be something that will need
+ * maintenance over time.  Since the profile list is embedded in
+ * libannocheck, there's no easy way to discover changes.  The
+ * assumption here is the profile naming scheme remains the same.
+ */
+static void set_libannocheck_profile(struct libannocheck_internals *anno, const char *product_release)
 {
-    size_t fl = 0;
-    size_t ll = 0;
-    char *workdir = NULL;
-    char *tmp = NULL;
+    const char **profiles = NULL;
+    unsigned int num_profiles = 0;
+    unsigned int i = 0;
+    libannocheck_error annoerr = 0;
 
-    assert(file != NULL);
+    if (anno == NULL || product_release == NULL) {
+        return;
+    }
 
-    if (s == NULL) {
-        return s;
+    /* get libannocheck profiles first */
+    annoerr = libannocheck_get_known_profiles(anno, &profiles, &num_profiles);
+
+    if (annoerr != libannocheck_error_none) {
+         warnx(_("libannocheck_get_known_profiles error: %s"), libannocheck_get_error_message(anno, annoerr));
+         return;
     }
 
-    fl = strlen(file->fullpath);
-    ll = strlen(file->localpath);
+    /* iterate over the profiles to try and find a match */
+    for (i = 0; i < num_profiles; i++) {
+        /* XXX: this needs a map in the config file */
```

Comment:
Do we need this map before merging?

Reply:
Good catch.  Forgot to finish that function.